gtsfm/bundle/bundle_adjustment.py

```diff
@@ -136,6 +145,37 @@ def create_computation_graph(self, sfm_data_graph: Delayed) -> Delayed:
         return dask.delayed(self.run)(sfm_data_graph)
 
 
+def aggregate_ba_stats(ba_data: GtsfmData) -> Dict[str,Any]:
```

Comment:
should this be a function in the `GtsfmData class`?

Reply:
Not sure. Seems specific to bundle adjustment, but I see your point too

Reply:
Seems specific to BA to me too, is GtsfmData internal to BA? Or is it more like the output of the whole pipeline?

Reply:
`GtsfmData` is the current data structure for 3d tracks + camera poses/parameters. We use it as output of DA, input to BA, output of BA, and input to MVS. So it is a more general structure.

Reply:
If we have it in `GtsfmData` we can reuse it in non-BA context too. For example: loading colmap results and publishing metrics for comparison.

Reply:
Thanks for the review, @ayushbaid 

That's a good point about being able to re-use the functionality for a COLMAP comparison. We should probably rename this function then, though, since GtsfmData is not necssarily a BA result. So we couldn't call it `aggregate_ba_stats()`

Maybe `aggregate_track_stats_dict()`?